daemon/src/main.rs

```diff
@@ -5,7 +5,7 @@ use std::io;
 use std::path::PathBuf;
 use tokio::signal;
 
-mod client;
+mod lsp_client;
```

Comment:
i'm not sure about the concrete name, but I felt like "client" is not specific enough :)

we have three kinds of clients in our codebase:
- the standalone "ethersync client" binary (now called `lsp_client`)
- the `editor` (representing an editor plugin connecting through LSP client and socket ), which is client to the daemon
- the `peer` (representing a fellow daemon to sync with) which is a client to the daemon